jgrapht-io/src/main/java/org/jgrapht/io/JSONExporter.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * (C) Copyright 2019-2019, Dimitrios Michail and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * See the CONTRIBUTORS.md file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the
+ * GNU Lesser General Public License v2.1 or later
+ * which is available at
+ * http://www.gnu.org/licenses/old-licenses/lgpl-2.1-standalone.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR LGPL-2.1-or-later
+ */
+package org.jgrapht.io;
+
+import java.io.PrintWriter;
+import java.io.Writer;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+import org.apache.commons.text.StringEscapeUtils;
+import org.jgrapht.Graph;
+
+/**
+ * Exports a graph to a JSON format.
```

Comment:
I'm not particularly well-versed in JSON, but is there like a standard which describes the format that we are exporting to? E.g. some RFC? And are we compliant with a specific JSON version? If so then we should probably mention these details here.

Reply:
Well, JSON is so common these days that there is no need for a reference to a document describing it. I can probably change the name to a href pointing to json.org.

If you mean about the exact schema that we are using here, no there is no standard for exporting graphs. I used a very simple one which is a list of nodes and a list of edges, which is compatible with the one used by the sigmajs Javascript library which contains graph drawing algorithms.

I will try improve the javadocs slightly.